core/synch.c

```diff
@@ -2289,3 +2289,201 @@ detach_on_permanent_stack(bool internal, bool do_cleanup, dr_stats_t *drstats)
     EXITING_DR();
     options_detach();
 }
+#ifdef LINUX
+void
+detach_externally_on_linux()
```

Comment:
How about "detach_externally_on_new_stack()" since we may use this same code on Mac and Android and maybe *BSD in the future and not just Linux.

Reply:
done